tensorboard/functionaltests/wct_test_driver.py

```diff
@@ -47,6 +50,30 @@ def create_test_class(binary_path, web_path):
     the test file's main module.
   """
 
+  class BrowserLogIndicatesResult(object):
+    def __init__(self):
+      self.passed = False
+    def __call__(self, driver):
+      # Scan through the log entries and search for a text indicating whether
+      # the test passed or failed.
+      messages = [entry['message'] for entry in driver.get_log('browser')]
+      if self._log_contains(messages, 'failing test'):
+        self.passed = False
+        return True
+      if self._log_contains(messages, 'test suite passed'):
+        self.passed = True
+        return True
+      # Here, we still don't know.
+      return False
+
+    def _log_contains(self, messages, text):
+      for message in messages:
+        if text in message:
+          # Print log message as an aid for debugging.
+          print("Found: %s in log entry message: %s" % (text, message))
```

Comment:
Note that the message is just `"%c1 failing test"`, so I’m not sure how
much this actually helps debugging (e.g., it doesn’t give you the name
of the failing test case), but this is fine with me.

Reply:
Good point. Changed to output entire log at end of test.